src/segment-loader.js

```diff
@@ -1448,18 +1490,33 @@ export default class SegmentLoader extends videojs.EventTarget {
       return;
     }
 
+    if (this.checkForIllegalMediaSwitch(trackInfo)) {
+      return;
+    }
+
+    trackInfo = trackInfo || {};
+
     // When we have track info, determine what media types this loader is dealing with.
     // Guard against cases where we're not getting track info at all until we are
     // certain that all streams will provide it.
-    if (typeof this.startingMedia_ === 'undefined' && (trackInfo.hasAudio || trackInfo.hasVideo)) {
-      this.startingMedia_ = trackInfo;
-    }
+    if ((trackInfo.hasVideo || trackInfo.hasAudio)) {
+      let changed = false;
 
-    this.trigger('trackinfo');
+      if (!this.startingMedia_) {
+        changed = true;
+        this.startingMedia_ = trackInfo;
+      }
+      if (!shallowEqual(this.currentMedia_, trackInfo)) {
+        this.currentMedia_ = trackInfo;
+        changed = true;
+      }
 
-    if (this.checkForIllegalMediaSwitch(trackInfo)) {
-      return;
+      if (changed) {
+        this.logger_('trackinfo update', trackInfo);
+        this.trigger('trackinfo');
```

Comment:
Would it be worth renaming to `newtrackinfo` for clarity?

Reply:
perhaps: `trackinfo` and `trackinfochange`. I don't know if we should other than to prevent breaking people that may have been using the event.